### PR TITLE
[add] tests refactoring per backend

### LINKED ISSUE #280: small memleak on RedisAI_ReplicateTensorSet
```
23935:signal-handler (1582162549) Received SIGTERM scheduling shutdown...
23935:M 20 Feb 2020 01:35:49.688 # User requested shutdown...
23935:M 20 Feb 2020 01:35:49.690 # Redis is now ready to exit, bye bye...
==23935== 
==23935== HEAP SUMMARY:
==23935==     in use at exit: 10,913,506 bytes in 212,812 blocks
==23935==   total heap usage: 1,262,259 allocs, 1,049,447 frees, 972,008,416 bytes allocated
==23935== 
==23935== 4 bytes in 1 blocks are definitely lost in loss record 42 of 125,246
==23935==    at 0x4C2FB0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==23935==    by 0x144D38: zmalloc (zmalloc.c:99)
==23935==    by 0x144F83: zstrdup (zmalloc.c:210)
==23935==    by 0x1C7006: RM_Strdup (module.c:352)
==23935==    by 0x16EF2140: RAI_ModelCreateTF (tensorflow.c:312)
==23935==    by 0x67E69A7: RAI_ModelCreate (model.c:195)
==23935==    by 0x67D784B: RedisAI_ModelSet_RedisCommand (redisai.c:620)
==23935==    by 0x1C76BE: RedisModuleCommandDispatcher (module.c:575)
==23935==    by 0x13D315: call (server.c:2468)
==23935==    by 0x13DFCA: processCommand (server.c:2763)
==23935==    by 0x14F086: processInputBuffer (networking.c:1470)
==23935==    by 0x14F1DB: processInputBufferAndReplicate (networking.c:1505)
==23935== 
==23935== 45 bytes in 2 blocks are definitely lost in loss record 84,708 of 125,246
==23935==    at 0x4C2FB0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==23935==    by 0x144D38: zmalloc (zmalloc.c:99)
==23935==    by 0x1531CF: createEmbeddedStringObject (object.c:85)
==23935==    by 0x15332B: createStringObject (object.c:121)
==23935==    by 0x1C81ED: RM_CreateString (module.c:904)
==23935==    by 0x1C838F: RM_CreateStringFromLongLong (module.c:943)
==23935==    by 0x67D8B57: RedisAI_ReplicateTensorSet (redisai.c:814)
==23935==    by 0x67D9395: RedisAI_Run_Reply (redisai.c:900)
==23935==    by 0x1CE072: moduleHandleBlockedClients (module.c:3895)
==23935==    by 0x13A84E: beforeSleep (server.c:1412)
==23935==    by 0x1344EB: aeMain (ae.c:500)
==23935==    by 0x141FD3: main (server.c:4234)
==23935== 
==23935== LEAK SUMMARY:
==23935==    definitely lost: 49 bytes in 3 blocks
==23935==    indirectly lost: 0 bytes in 0 blocks
==23935==      possibly lost: 59,069 bytes in 1,455 blocks
==23935==    still reachable: 2,801,134 bytes in 65,299 blocks
==23935==                       of which reachable via heuristic:
==23935==                         stdstring          : 2,230,900 bytes in 56,726 blocks
==23935==                         newarray           : 4,480 bytes in 10 blocks
==23935==         suppressed: 8,053,254 bytes in 146,055 blocks
==23935== Reachable blocks (those to which a pointer was found) are not shown.
==23935== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==23935== 
==23935== For counts of detected and suppressed errors, rerun with: -v
==23935== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: 1 from 1)
	bad exit code for serverId 1
	Redis did not exit cleanly
basic_tests:test_run_mobilenet_multiproc
```

### LINKED ISSUE #285: memleak on torchLoadModel
```
==28935== 48 bytes in 1 blocks are definitely lost in loss record 95,420 of 132,320
==28935==    at 0x4C3017F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==28935==    by 0x6A8A814: torchLoadModel (torch_c.cpp:323)
==28935==    by 0x6A7A0E5: RAI_ModelCreateTorch (torch.c:42)
==28935==    by 0x67E6B05: RAI_ModelCreate (model.c:209)
==28935==    by 0x67D784B: RedisAI_ModelSet_RedisCommand (redisai.c:620)
==28935==    by 0x1C76BE: RedisModuleCommandDispatcher (module.c:575)
==28935==    by 0x13D315: call (server.c:2468)
==28935==    by 0x13DFCA: processCommand (server.c:2763)
==28935==    by 0x14F086: processInputBuffer (networking.c:1470)
==28935==    by 0x14F1DB: processInputBufferAndReplicate (networking.c:1505)
==28935==    by 0x14F551: readQueryFromClient (networking.c:1587)
==28935==    by 0x1342E2: aeProcessEvents (ae.c:443)

```

### PR DESCRIPTION
**Pull request by filipecosta90**
@lantiga and @rafie this splits the tests per backend and the common ones. It also extends coverage and ensures slave sync when required. The whole purpose is to clean and make the tests clear and remove code duplicates 
-  test/includes.py - has the methods to load models, check the env variables, etc... 
-  test/tests_common.py  - tests for common commands like tensorset 
-  test/tests_onnx.py - tests for ONNX backend
-  test/tests_pytorch.py  - tests for torch backend
-  test/tests_tensorflow.py  - tests for TF backend
-  test/tests_tflite.py  - tests for tflite backend

leak fixes:
- fixes #280 
- fixes #285

### CODE REVIEW

File: test/tests_pytorch.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    env.assertEqual(type(exception), redis.exceptions.ResponseError)
+    env.assertEqual("no script at key", exception.__str__())
+
+    # ERR wrong type from SCRIPTDEL
+    try:
+        con.execute_command('SET', 'NOT_SCRIPT', 'BAR')
+        con.execute_command('AI.SCRIPTDEL', 'NOT_SCRIPT')
+    except Exception as e:
+        exception = e
+    env.assertEqual(type(exception), redis.exceptions.ResponseError)
```
Comment (rafie) [inline test/tests_pytorch.py]: inside except block?

File: test/tests_pytorch.py [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+    env.assertFalse(con.execute_command('EXISTS', 'ket'))
+
+    # ERR no script at key from SCRIPTDEL
+    try:
+        con.execute_command('DEL', 'EMPTY')
+        con.execute_command('AI.SCRIPTDEL', 'EMPTY')
+    except Exception as e:
+        exception = e
+    env.assertEqual(type(exception), redis.exceptions.ResponseError)
```
Comment (rafie) [inline test/tests_pytorch.py]: inside except block?

File: opt/Makefile [hunk 3]
```diff
@@ -142,42 +142,44 @@ endif
...
 
+TEST_CMD=DEVICE=$(DEVICE) PYDEBUG=$(PYDEBUG) python3 -m RLTest $(TEST_ARGS) --module $(INSTALL_DIR)/redisai.so
+
 TEST_REPORT_DIR ?= $(PWD)
 ifeq ($(VERBOSE),1)
 TEST_ARGS += -v
 endif
 ifeq ($(TEST),)
-TEST=basic_tests.py
+TEST=tests_common.py tests_pytorch.py tests_onnx.py tests_tflite.py tests_tensorflow.py
```
Comment (rafie) [inline opt/Makefile]: The $(TEST) variable is there for picking a single test for debugging purposes (in a file.py:specific_test form). For execution of several tests, I bet RLTest can run all tests in a given dir.

Comment (filipecosta90) [reply opt/Makefile]: @rafie does this alter the behaviour?
```
make -C opt test TEST="tests_common.py:test_set_tensor  tests_pytorch.py:test_run_torch_model"
```
would still be valid right? ( testing it )

Comment (rafie) [reply opt/Makefile]: Well, from a second look, my design seems to be wrong here. I meant (as it is in other modules), that TEST would be given something that would match the RLTEST --test argument, and then RLTest be invoked with -s (for output from Redis) and PYDEBUG set for python debugging. Instead, the Makefile would override TEST with basic_test.py (which was the only test file). I think it's better to let the user explicitly specify tests (like you do in your example), and not pass --tests $(TEST) to RLTest at all if TEST is not set. Something like:
```
ifneq ($(TEST),)
TEST_ARGS += --test $(TEST)
emdif
```

File: test/tests_pytorch.py (hunk 1 above)
Comment (filipecosta90) [reply test/tests_pytorch.py]: fixed

File: test/tests_pytorch.py (hunk 2 above)
Comment (filipecosta90) [reply test/tests_pytorch.py]: fixed :)

**Pull request merged**